### PR TITLE
Prune control characters from Xml CDATA

### LINKED ISSUE #1566: Invalid XML characters in Params in testng-results.xml
### TestNG Version
6.11

### Expected behavior
Invalid XML 1.0 characters are escaped, replaced, or removed from the output. Or the file being XML 1.1 which supports more characters

### Actual behavior
The 0xC character is present in the testng-results.xml file, making in unopenable in other tools like Jenkins

### Is the issue reproducible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
```java
@DataProvider
public Object[][] dataProvider() {
	return new Object[][]{{
		"test \u000C"
	}};
}

@Test(dataProvider = "dataProvider")
public void test(String argument) {
}
```

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1566

Fixes #1566.

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/reporters/XMLStringBuffer.java [hunk 1]
```diff
@@ -305,6 +305,9 @@ public void setDefaultComment(String defaultComment) {
    * Add a CDATA tag.
    */
   public void addCDATA(String content) {
+    if (content != null) {
+        content = content.replaceAll("\\p{Cc}", "");
```
Comment (juherr) [inline src/main/java/org/testng/reporters/XMLStringBuffer.java]: If possible, `\u000C` should be replaced by `&#xc;`:
http://www.fileformat.info/info/unicode/char/000c/index.htm

**Review by juherr — APPROVED** [review]
Looks good 👍

File: src/main/java/org/testng/reporters/XMLStringBuffer.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/reporters/XMLStringBuffer.java]: I am basically using the regex pattern which stands for control characters and I believe `\u000C` is a control character (Form feed)

Comment (juherr) [reply src/main/java/org/testng/reporters/XMLStringBuffer.java]: Yes, but we are loosing information. Is it possible to do better?
If no, it is not a problem :)

Comment (krmahadevan) [reply src/main/java/org/testng/reporters/XMLStringBuffer.java]: What sort of information are you referring to as being lost ? I didn't get that part. Can you please elaborate ?

Comment (juherr) [reply src/main/java/org/testng/reporters/XMLStringBuffer.java]: `"&#xc;"` and `""` are not the same thing.
But I agree the difference is not important and can be skipped.

**Comment by krmahadevan** [comment]
ping @cbeust - Can you please help with merging this ?

**Pull request merged**